pkg/s3select/sql/record.go

```diff
@@ -36,8 +40,8 @@ const (
 type Record interface {
 	Get(name string) (*Value, error)
 	Set(name string, value *Value) error
-	MarshalCSV(fieldDelimiter rune) ([]byte, error)
-	MarshalJSON() ([]byte, error)
+	MarshalCSV(w io.Writer, fieldDelimiter rune) error
+	MarshalJSON(w io.Writer) error
```

Comment:
`Marshal*()` methods return `([]byte, error)` like go standard library. If we want to write to given `io.Writer`, then we would need to rename this to `Write*()`.

Reply:
changed

Reply:
What about Encode*() ?  Encode*() is more specific~